ecdsa/party/implementation.go

```diff
@@ -305,9 +313,23 @@ func (p *Impl) authoriseSignerToSign(digest Digest, signer *singleSigner) error
 	defer signer.mtx.Unlock()
 
 	if signer.localParty == nil {
+		randomnessSeed := append(p.partiesSharedSecret, digest[:]...)
+		parties, err := shuffleParties(randomnessSeed, p.parameters.Parties().IDs())
+		if err != nil {
+			return err
+		}
+
+		parties = tss.SortPartyIDs(parties[:p.parameters.Threshold()+1])
```

Comment:
we're using p.parameters.Threshold()+1 everywhere, right? maybe just capture the +1 inside the threshold to simplify.